app/lib/katello/resources/cdn.rb

```diff
@@ -24,9 +29,9 @@ def substitutor
         end
 
         def initialize(url, options = {})
-          @ssl_version = Setting[:cdn_ssl_version]
-          if @ssl_version && !SUPPORTED_SSL_VERSIONS.include?(@ssl_version)
-            fail("Invalid SSL version specified. Check the 'CDN SSL Version' setting")
+          @cdn_min_tls_version = Setting[:cdn_min_tls_version]
+          if @cdn_min_tls_version && !SUPPORTED_TLS_VERSIONS.include?(@cdn_min_tls_version)
+            fail("Invalid TLS min_version specified. Check the 'CDN Minimum Allowed TLS Version' setting")
```

Comment:
How will users see this failure? Does it propagate to the UI and API?

Reply:
The other tricky part being that an organization admin will run into this issue and then have to contact a Satellite admin in order to modify this. I realize that's how the current functionality already works, but my point being we should make sure this is as actionable by a user encountering it as possible.

Reply:
`fail()` should cause an ISE. I'm not sure if one would need to look at production.log to see the exact message.

That said, I think the only way this condition could be met is if the user has a value already set, and then that value later becomes unavailable (an upgrade to ruby or to the operating system removes the currently configured TLS version).

As long as we test on all supported platforms, I think our testing would catch that and appropriate intervention would be made before an end user would ever face that issue.

I kept and updated the check from the previous implementation (which seems also unlikely to be reached in real world usage) just in case.

If you still have concerns about it, maybe we can create a new issue to follow up? Let me know what you think, thanks.